.gitignore

```diff
@@ -79,8 +79,9 @@ sklearn/utils/_seq_dataset.pxd
 sklearn/linear_model/_sag_fast.pyx
 
 # deprecated paths
-# TODO: Remove in 0.24
-# All of these files should have a match in _build_utils/deprecated_modules.py
+# All of these files were deprecated and automatically generated in 0.22, and
+# removed in 0.24. We keep them in gitignore otherwise devs would need to
+# manually delete them
```

Comment:
`git clean -xfd` does that though. I'd rather remove them from here and have us run the command once?

Reply:
Ok, I'm fine with either.

We should probably recommend `git clean -xi` since this will remove all untracked files. We can do that in a sticky issue I guess?

Pinging @glemaitre @rth @thomasjpfan @ogrisel @jnothman , any objection?

Reply:
I would prefer to remove them from `.gitignore`.

Although, I do not know where to put the `git clean` information. We would  most likely need to help contributors with this when they sync with master and then use `git add .` which would add these ignored files back.

Reply:
thoughts on a sticky issue as suggested?

Reply:
Not sure what they are, but I like the sound of it

Reply:
We can pin issues to the top as done in https://github.com/Microsoft/LightGBM/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

Reply:
We didn't get a chance to discuss this during the meeting so I'll merge tomorrow unless someone objects